spec/views/users/show.html.erb_spec.rb

```diff
@@ -0,0 +1,73 @@
+  before(:each) do
+    @user1 = User.create(name: 'Peter Parker', bio: 'Official spiderman hotographer', photo: 'https://www.example.com',
+                         posts_counter: 0)
+
+    @user2 = User.create(name: 'Bruce Wayne', bio: 'Millionaire', photo: 'https://www.example.com', posts_counter: 3)
+
+    @post1 = Post.create(author_id: @user1.id, title: 'First post', text: 'This is my first post', comments_counter: 0,
+                         likes_counter: 0)
+    @post2 = Post.create(author_id: @user1.id, title: 'Second post', text: 'This is my second post',
+                         comments_counter: 0,
+                         likes_counter: 0)
+    @post3 = Post.create(author_id: @user1.id, title: 'Third post', text: 'This is my third post', comments_counter: 0,
+                         likes_counter: 0)
+    @post4 = Post.create(author_id: @user1.id, title: 'Fourth post', text: 'This is my fourth post',
+                         comments_counter: 0,
+                         likes_counter: 0)
+
+    # Second user single post
+    @post5 = Post.create(author_id: @user2.id, title: 'Hello World', text: 'My first post too!', comments_counter: 0,
+                         likes_counter: 0)
+
+    # First post interactions
+    Comment.create(post_id: @post1.id, author_id: @user1.id, text: 'First comment! Wow')
+    Comment.create(post_id: @post1.id, author_id: @user2.id, text: 'Nice post BTW')
+    Comment.create(post_id: @post1.id, author_id: @user2.id, text: 'Yeah I agree')
+
+    Like.create(post_id: @post1.id, author_id: @user2.id)
+
+    # Second post interactions
+    Comment.create(post_id: @post2.id, author_id: @user1.id, text: 'First comment! Wow')
+
+    Like.create(post_id: @post2.id, author_id: @user1.id)
+  end
+
+  describe 'user show page' do
+    it 'displays the right username' do
+      visit user_path(@user1)
+      expect(page).to have_content('Peter Parker')
+      expect(page).to_not have_content('Bruce Wayne')
+    end
+
+    it 'displays the users profile photo' do
+      visit user_path(@user1)
+      expect(page).to have_css("img[src*='https://www.example.com']")
+    end
+
+    it 'displays the right biography of first user' do
+      visit user_path(@user1)
+      expect(page).to have_content('Official spiderman hotographer')
+      expect(page).to_not have_content('Millionaire')
+    end
+
+    it 'displays the right biography of second user' do
+      visit user_path(@user2)
+      expect(page).to have_content('Millionaire')
+      expect(page).to_not have_content('Official spiderman hotographer')
+    end
+
+    it 'displays the number of posts the first user has written' do
+      visit user_path(@user1)
+      expect(page).to have_content('Number of posts: 4')
+    end
+
+    it 'displays the last 3 posts' do
+      visit user_path(@user1)
+      expect(page).to have_content('This is my third post')
+      expect(page).to have_content('This is my second post')
+    end
+  end
+end
```

Comment:
-  ### For users show page you should also have the following tests:
- [ ] Asserts you can see a button that lets me view all of a user's posts
- [ ] Tests that redirects to the post's show page when you click on a user's post
- [ ] Tests that redirects to the user's post's index page when you click on a see all posts
![image](https://user-images.githubusercontent.com/62705947/215595482-9d910c47-fb23-4fd1-8163-d7302dc44adc.png)